src/plugins/discover/public/application/main/utils/use_saved_search.ts

```diff
@@ -205,7 +205,11 @@ export const useSavedSearch = ({
       }
     });
 
-    return () => subscription.unsubscribe();
+    const abortController = refs.current.abortController;
+    return () => {
+      abortController?.abort();
+      subscription.unsubscribe();
+    };
```

Comment:
```suggestion
    return () => {
      refs.current.abortController?.abort();
      subscription.unsubscribe();
    };
```

Reply:
So you're storing a reference to the first abortController assigned, however, we assign newer abortController instances on every fetch triggered
https://github.com/elastic/kibana/blob/12c1b467893578a3d50c1718746ccc10949858b9/src/plugins/discover/public/application/main/utils/use_saved_search.ts#L181-L185
With this code change it should work 
I've tested by searching with a lucene regular expression like `/.*a.*x.*/`, and my navigating to dashboard, and it worked:
![Bildschirmfoto 2022-04-07 um 10 28 07](https://user-images.githubusercontent.com/463851/162156546-d13de51c-8c7f-457b-98d1-3ced92f5cb05.png)
Applied some changes to the a-la-carte instances using vim to debug it

Reply:
Thx for the catch, tested on `a-la-carte` and it now works as expected. 

I would suggest to remove `abortController` reference from react `ref.current`, since anyway we are not exporting it from the `useSavedSearch` hook. Local variable usage will also meet react rules in clean up function.

![D30B6FED-4D06-4FD0-B341-16729DC53B35_4_5005_c](https://user-images.githubusercontent.com/39378793/162422116-9bed5557-0e60-432e-ae95-888ca1148bed.jpeg)